newsfragments/670.feature.rst

```diff
@@ -0,0 +1 @@
+* Reworked :mod:`trio.socket` namespace construction, making it more understandable by static analysis tools. This should improve tab completion in editors, reduce false positives from pylint, and is another step towards providing type hints.
```

Comment:
Thinking about it, it's going to be kind of weird-looking in our final [change log](https://trio.readthedocs.io/en/latest/history.html) to have multiple almost-identical entries with just the module name changed.

Since #542 is already an umbrella bug for these changes, how about we drop this file, and instead modify the existing `542.feature.rst` to cover the new work as well? And while we're at it, we can add mention of the work in #656 to fix `trio.testing`... so it can say something like:

"Reworked :mod:`trio`, :mod:`trio.testing`, and :mod:`trio.socket` namespace construction, making them more understandable..."

Also BTW, newsfragments shouldn't start with a `*`. Towncrier will automatically add a `*` when it assembles the release notes, and then things will look weird if we have two `*`s :-)

Reply:
Yes, makes sense. I'll correct this later this day, currently commuting. I actually put this in in haste to have a change, before I learned that I could actually do an empty commit.